src/db/graph/queries/get.rs

```diff
@@ -362,28 +363,108 @@ pub fn get_global_hot_tags_taggers(tag_list: &[&str]) -> Query {
     .param("labels", tag_list)
 }
 
-// Analyzes tag usage for a specific list of user IDs. Groups tags by name,
-// showing for each: label, post count and list of user IDs
-// Orders by post_count (descending).
-// Note: Only considers users from the provided users_id list.
-pub fn get_tags_by_user_ids(users_id: &[&str]) -> Query {
+pub fn get_tag_taggers_by_reach(
+    label: &str,
+    user_id: &str,
+    reach_subquery: String,
+    skip: usize,
+    limit: usize,
+) -> Query {
     query(
-        "
-        UNWIND $ids AS id
-        MATCH (u:User)-[tag:TAGGED]->(p:Post)
-        WHERE u.id = id
-        WITH tag.label AS label, COLLECT(DISTINCT u.id) AS taggers, COUNT(DISTINCT p) AS uniquePosts
-        WITH {
+        format!(
+            "
+            {}
+            MATCH (reach)-[tag:TAGGED]->()
+            WHERE user.id = $user_id AND tag.label = $label
+            WITH reach, MAX(tag.indexed_at) AS tag_time
+            ORDER BY tag_time DESC
+            SKIP $skip LIMIT $limit
+            RETURN COLLECT(reach.id) as tagger_ids
+            ",
+            reach_subquery
+        )
+        .as_str(),
+    )
+    .param("label", label)
+    .param("user_id", user_id)
+    .param("skip", skip as i64)
+    .param("limit", limit as i64)
+}
+
+pub fn get_hot_tags_by_reach(
+    user_id: &str,
+    reach_subquery: String,
+    tags_query: &HotTagsInput,
+) -> Query {
+    let input_tagged_type = match &tags_query.tagged_type {
+        Some(tagged_type) => tagged_type.to_string(),
+        None => String::from("Post|User"),
+    };
+    query(
+        format!(
+            "
+        {}
+        MATCH (reach)-[tag:TAGGED]->(tagged:{})
+        WHERE user.id = $user_id AND tag.indexed_at >= $from AND tag.indexed_at < $to
```

Comment:
As far as we've learned this is computationally expensive... but, whatever (for now) :sweat_smile:

Reply:
if it becomes heavy, we can add special indices in the graph for it.

Reply:
Let's add them or add a `//TODO: ` so we are mindful we are doing this here.